clang/include/clang/Driver/Options.td

```diff
@@ -6910,6 +6910,10 @@ defm loop_versioning : BoolOptionWithoutMarshalling<"f", "version-loops-for-stri
 
 def fhermetic_module_files : Flag<["-"], "fhermetic-module-files">, Group<f_Group>,
   HelpText<"Emit hermetic module files (no nested USE association)">;
+
+def do_concurrent_parallel_EQ : Joined<["-"], "fdo-concurrent-parallel=">,
```

Comment:
Can you make the option name reflect that this is targeting OpenMP?

Reply:
Great suggestion @clementval -- especially because there could someday be alternative options such as OpenACC for example.

Reply:
Yes there will be an OpenACC version of this.

Reply:
Changed to `do-concurrent-to-openmp`. Let me know if the name can be improved.

Reply:
Shall we have a single option `-fdo-concurrent-parallel=` and have the decision of whether to use OpenMP or OpenACC via `-fopenmp` and `-fopenacc`?  `-fdo-concurrent-parallel` does not make much sense without one of the latter options. I guess it's a matter of proper documentation.  If we have different flavors of this option, we would run into error cases that we'd have to handle such as `-fopenacc -fdo-concurrent-to-openmp`.

Reply:
PS: Does `-fdo-concurrent-parallel=` issue an error if actual parallelism is requested and `-fopenmp` is not present?

PPS: If we choose to rename the thing, then is should rather be `-fopenmp-parallel-do-concurrent=`, since this seems to be the common naming pattern for OpenMP-related compiler flags.

Reply:
I somewhat agree with @mjklemm regarding the name (that's why I didn't put `openmp` at the start). However, I will leave it for the reviewers. Also, this is something that I hope will be easy to fix retrospectively if we want to; let me know if you disagree.

> PS: Does -fdo-concurrent-parallel= issue an error if actual parallelism is requested and -fopenmp is not present?

Yes, see changes in `FrontendActions.cpp`, an error message is emitted.

> PPS: If we choose to rename the thing, then is should rather be -fopenmp-parallel-do-concurrent=, since this seems to be the common naming pattern for OpenMP-related compiler flags.

In my mind, this is a `do concurrent` feature/flag not an OpenMP one. It just happens to map to OpenMP atm.